pkg/archive/wrap.go

```diff
@@ -22,7 +23,7 @@ import (
 //
 // FIXME: stream content instead of buffering
 // FIXME: specify permissions and other archive metadata
-func Generate(input ...string) (Archive, error) {
+func Generate(input ...string) (io.ReadCloser, error) {
```

Comment:
@aaronlehmann I'm worried making more changes will cause this to spiral further. I don't quite understand what this is actually doing, so let's keep the changes to a minimum.